### DISCUSSION TITLE
When restoring a backup, does ludusavi uses the mapping.yaml file to find the right path to restore the files?

### DISCUSSION THREAD

**Discussion by Dr4w** [mtkennerly/ludusavi#321] [Q&A] [answered]
As title, does ludosavi store the correct path to restores the files in the mapping.yaml file? Or does it also uses some settings stored locally?
I'd like to know so that in case of changing the PC of which ludusavi is installed on, I would need to set up all the paths manually or not to restore the backup correctly. Thanks.

**Comment by mtkennerly** [comment, marked as answer]
Hi!

Ludusavi stores the full path to each file in the backup (mapping.yaml) and will restore to the same path by default. For example, if you have a save file at `C:/Users/sys1-name/some-game/save.dat`, then the backup will include your username in the path.

When restoring  on another system, you can go to the "other" screen and configure a redirect to restore to the new username/etc. There's more info about redirects here: https://github.com/mtkennerly/ludusavi?tab=readme-ov-file#redirects

In the above example, you could set up a restore redirect like this, to say that any backed up file from `sys1-name`'s user folder should be restored to `sys2-name` instead:

> ![image](https://github.com/mtkennerly/ludusavi/assets/4934192/984f95f8-f544-4e96-aef2-6fab73c6b734)

More broadly, if you play on multiple systems often and want to "combine" backups from your user folder  on all systems, then you could use a bidirectional redirect with a dummy path. This is a bit more advanced. On system 1, you'd set up a bidirectional redirect like this:

> ![image](https://github.com/mtkennerly/ludusavi/assets/4934192/bd46d340-722c-4c04-873a-08dd795902c9)

`X:\user` doesn't really exist, but that's how the backup will identify the file, and it'll still get restored to `sys1-name`. Then on system 2, you'd do the same, except put `sys2-name` in the first path instead.

**Reply by Dr4w** [reply]
I see, thanks. So, by default, if I don't setup a redirect but just restore a backup on another system, if this system is using the same User folder and the path of the saves is the same as the original system where the backup was performed, it will be restored following the original backup path?

**Reply by mtkennerly** [reply]
That's correct :)
